poe-tasks/detect-python-cdk.py

```diff
@@ -0,0 +1,210 @@
+
+        if cdk_info["_raw_type"] == "git":
+            git_url = cdk_info.get("git", "unknown")
+            git_ref = cdk_info.get("branch", cdk_info.get("rev", "unknown"))
+            print(f"   Issue: Git reference: {git_url}#{git_ref}")
+        elif cdk_info["_raw_type"] == "path":
+            local_path = cdk_info.get("path", "unknown")
+            print(f"   Issue: Local path reference: {local_path}")
+        elif cdk_info["_raw_type"] == "url":
+            url = cdk_info.get("url", "unknown")
+            print(f"   Issue: URL reference: {url}")
+        elif cdk_info["_raw_type"] == "unknown":
+            print(f"   Issue: Non-standard version string: {cdk_info.get('version', 'unknown')}")
+        else:
+            print(f"   Issue: Unexpected dependency format: {cdk_info.get('_raw_dependency', 'unknown')}")
+
+        print()
+        print("   It is currently pinning its CDK version to a local or git-based ref.")
+        print("   To resolve, use `poe use-cdk-latest` after your working dev version")
+        print("   of the CDK has been published.")
+        if connector_name:
+            print(f"   You can also use the slash command in your PR: `/poe connector {connector_name} use-cdk-latest`")
+
+        return False
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Detect and analyze airbyte-cdk dependency information", formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+
+    parser.add_argument("directory", nargs="?", default=".", help="Directory containing pyproject.toml (default: current directory)")
+
+    mode_group = parser.add_mutually_exclusive_group()
+    mode_group.add_argument("--extras-only", action="store_true", help="Return extras string for poetry add command")
+    mode_group.add_argument("--verify-version-pin", action="store_true", help="Verify CDK is pinned to standard version (exit 1 if not)")
+
+    args = parser.parse_args()
+
+    connector_dir = Path(args.directory)
+    pyproject_path = connector_dir / "pyproject.toml"
+
+    if not pyproject_path.exists():
+        if args.extras_only:
+            return
+        elif args.verify_version_pin:
+            print(f"Error: pyproject.toml not found in {connector_dir}")
+            sys.exit(1)
+        else:
+            print(json.dumps({"error": f"pyproject.toml not found in {connector_dir}"}))
+            return
+
+    cdk_info = parse_cdk_dependency(pyproject_path)
+
+    if args.extras_only:
+        extras = cdk_info.get("extras", [])
+        print(format_extras_for_poetry(extras))
+
+    elif args.verify_version_pin:
+        resolved_dir = connector_dir.resolve()
+        connector_name = None
+
+        if resolved_dir.name.startswith(("source-", "destination-")):
+            connector_name = resolved_dir.name
+        elif not resolved_dir.name.startswith(("source-", "destination-")):
+            print(f"Warning: Directory '{resolved_dir.name}' doesn't start with 'source-' or 'destination-'")
+
```

Comment:
I'll commit this simplification. Pull the latest to get the update.

```suggestion
        success = cdk_info["_is_production_ready"]

```